app/controllers/chargeback_controller.rb

```diff
@@ -288,28 +288,32 @@ def cb_assign_field_changed
 
   # Add a new tier at the end
   def cb_tier_add
-    detail_index = params[:detail_index]
+    index = params[:index]
+    detail_index, tier_index = index.split("-")
+    tier_index = tier_index.to_i
     ii = detail_index.to_i
 
     @edit  = session[:edit]
     detail = @edit[:new][:details][ii]
-
     @edit[:new][:num_tiers][ii] = detail[:chargeback_tiers].to_a.length if detail[:chargeback_tiers]
     @edit[:new][:num_tiers][ii] = 1 unless @edit[:new][:num_tiers][ii] || @edit[:new][:num_tiers][ii] == 0
     @edit[:new][:num_tiers][ii] += 1
 
-    tier_index = @edit[:new][:num_tiers][ii] - 1
     tier_list = @edit[:new][:tiers][ii]
-    tier_list[tier_index] = {}
 
-    tier                 = tier_list[tier_index]
-    tier[:start]         = tier_list[tier_index - 1][:finish]
-    tier[:finish]        = Float::INFINITY
+    tier                 = {}
+    tier[:start]         = tier_list[tier_index][:finish]
+    tier[:finish] = if tier_list[tier_index + 1].nil?
```

Comment:
there can be ternary operator
`tier_list[tier_index + 1].nil? ? Float::INFINITY : tier_list[tier_index + 1][:start]`